app/kuma-dp/pkg/dataplane/metrics/server_test.go

```diff
@@ -173,40 +173,30 @@ var _ = Describe("Process Metrics", func() {
 
 var _ = Describe("ProcessNewlineChars", func() {
```

Comment:
Since this function is only used inside `processMetrics` I think we can test the `processNewlineChars` functionality as part of the `processMetrics` tests

Reply:
Yes, we can. But `ProcessMetrics` does multiple things, where many could go wrong and it also takes input and expected from `testdata/` which might be inconvenient to test small functions like `processNewlineChars`. So this test ensures it does that right and it's very easy to analyse as the input and expected data are small/simple.
But I think `dedup` test can be removed, as at this moment testcases of `dedup` and `processNewlineChars` are overlapping.

Reply:
> it also takes input and expected from `testdata/` which might be inconvenient to test small functions like `processNewlineChars`

Hmm the behavior of `processNewlineChars` isn't important per se, we really only care about making sure our metrics are processed, not arbitrary input, since we don't export the function.
Do the `processMetrics` test cases already test this behavior around deduplicating new lines?

Reply:
yes, most of the cases are covered by `processMetrics` test cases, except `\n` or `space` characters in the beginning of a the string data.
Ex: `\n\nMERGED_APP1_APP2_METRICS`

Reply:
Added the above mentioned case to the existing test covered by `processMetrics` and removed test for `processNewlineChars`.